packages/core-flows/src/price-list/steps/update-price-list-prices.ts

```diff
@@ -0,0 +1,90 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import {
+  IPricingModuleService,
+  PriceSetMoneyAmountDTO,
+  UpdatePriceListPriceDTO,
+  UpdatePriceListPricesDTO,
+  UpdatePriceListPricesWorkflowDTO,
+} from "@medusajs/types"
+import { buildPriceSetPricesForModule } from "@medusajs/utils"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+type WorkflowStepInput = {
+  data?: UpdatePriceListPricesWorkflowDTO[]
+  variant_price_map: Record<string, string>
+}
+
+export const updatePriceListPricesStepId = "update-price-list-prices"
+export const updatePriceListPricesStep = createStep(
+  updatePriceListPricesStepId,
+  async (stepInput: WorkflowStepInput, { container }) => {
+    const { data = [], variant_price_map: variantPriceSetMap } = stepInput
+    const priceListPricesToUpdate: UpdatePriceListPricesDTO[] = []
+    const pricingModule = container.resolve<IPricingModuleService>(
+      ModuleRegistrationName.PRICING
+    )
+
+    for (const priceListData of data) {
+      const pricesToUpdate: UpdatePriceListPriceDTO[] = []
+      const { prices = [], id } = priceListData
+
+      for (const price of prices) {
+        pricesToUpdate.push({
+          ...price,
+          price_set_id: variantPriceSetMap[price.variant_id!],
+        })
+      }
+
+      priceListPricesToUpdate.push({
+        price_list_id: id,
+        prices: pricesToUpdate,
+      })
+    }
+
+    const priceIds = priceListPricesToUpdate
+      .map((priceListData) => priceListData.prices.map((price) => price.id))
+      .flat(1)
+      .filter(Boolean)
```

Comment:
suggestion: Move both the mapping and filtering to the loop just before:

```ts
let priceIds: string[] = []
for (const priceListData of data) {
  const pricesToUpdate: UpdatePriceListPriceDTO[] = []
  const { prices = [], id } = priceListData

  for (const price of prices) {
    pricesToUpdate.push({
      ...price,
      price_set_id: variantPriceSetMap[price.variant_id!],
    })

    // This should take care of both no?
    if (price.id) {
      priceIds.push(price.id)
    }
  }

  priceListPricesToUpdate.push({
    price_list_id: id,
    prices: pricesToUpdate,
  })
}
```

Reply:
nice!